internal/sirius/get_my_permissions_test.go

```diff
@@ -0,0 +1,225 @@
+		PactFileWriteMode: "merge",
+		LogDir:            "../../logs",
+		PactDir:           "../../pacts",
+	}
+	defer pact.Teardown()
+
+	testCases := []struct {
+		name             string
+		setup            func()
+		cookies          []*http.Cookie
+		expectedResponse PermissionSet
+		expectedError    error
+	}{
+		{
+			name: "OK",
+			setup: func() {
+				pact.
+					AddInteraction().
+					Given("User exists").
+					UponReceiving("A request to get my permissions").
+					WithRequest(dsl.Request{
+						Method: http.MethodGet,
+						Path:   dsl.String("/api/permission"),
+						Headers: dsl.MapMatcher{
+							"X-XSRF-TOKEN":        dsl.String("abcde"),
+							"Cookie":              dsl.String("XSRF-TOKEN=abcde; Other=other"),
+							"OPG-Bypass-Membrane": dsl.String("1"),
+						},
+					}).
+					WillRespondWith(dsl.Response{
+						Status:  http.StatusOK,
+						Headers: dsl.MapMatcher{"Content-Type": dsl.String("application/json")},
+						Body: dsl.Like(map[string]interface{}{
+							"data": map[string]interface{}{
+								"user": map[string]interface{}{
+									"permissions": dsl.EachLike("PATCH", 1),
+								},
+								"team": map[string]interface{}{
+									"permissions": dsl.EachLike("POST", 1),
+								},
+							},
+						}),
+					})
+			},
+			cookies: []*http.Cookie{
+				{Name: "XSRF-TOKEN", Value: "abcde"},
+				{Name: "Other", Value: "other"},
+			},
+			expectedResponse: PermissionSet{
+				"user": PermissionGroup{Permissions: []string{"PATCH"}},
+				"team": PermissionGroup{Permissions: []string{"POST"}},
+			},
+		},
+		{
+			name: "Unauthorized",
+			setup: func() {
+				pact.
+					AddInteraction().
+					Given("User exists").
+					UponReceiving("A request to get my permissions without cookies").
+					WithRequest(dsl.Request{
+						Method: http.MethodGet,
+						Path:   dsl.String("/api/permission"),
+						Headers: dsl.MapMatcher{
+							"OPG-Bypass-Membrane": dsl.String("1"),
+						},
+					}).
+					WillRespondWith(dsl.Response{
+						Status: http.StatusUnauthorized,
+					})
+			},
+			expectedError: ErrUnauthorized,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.setup()
+
+			assert.Nil(t, pact.Verify(func() error {
+				client, _ := NewClient(http.DefaultClient, fmt.Sprintf("http://localhost:%d", pact.Server.Port))
+
+				myPermissions, err := client.GetMyPermissions(getContext(tc.cookies))
+				assert.Equal(t, tc.expectedResponse, myPermissions)
+				assert.Equal(t, tc.expectedError, err)
+				return nil
+			}))
+		})
+	}
+}
+
+func TestPermissionsIgnoredPact(t *testing.T) {
+	// We need this test to produce a specific array of permissions in the
+	// response so that Cypress tests will pass. Since Pact won't let us return
+	// multiple array entries from `dsl.EachLike` we have to write a separate
+	// test with the specific output. We call the consumer "z-append" to ensure
+	// the Pact is loaded last and overwrites the actual test of /api/permission
```

Comment:
Interesting way of doing this, I'm not sure how I feel about using a test to overwrite mock data. I guess the pro is that you know the code handles this case our side... 🤔 

Maybe I should have done the teams one like this

Reply:
I tried a few approaches and most of them just aren't supported by Pact. I suspect the proper answer is to not rely on _just_ Pact for all our integration tests, but that model works for us now and I don't want to mess with it.